src/core_plugins/kibana/public/context/app.js

```diff
@@ -0,0 +1,100 @@
+import _ from 'lodash';
+
+import uiModules from 'ui/modules';
+import contextAppTemplate from './app.html';
+import './components/loading_button';
+import './components/size_picker/size_picker';
+import {
+  createInitialQueryParametersState,
+  QueryParameterActionsProvider,
+  QUERY_PARAMETER_KEYS,
+} from './query_parameters';
+import { QueryActionsProvider } from './query';
+
+const module = uiModules.get('apps/context', [
+  'elasticsearch',
+  'kibana',
+  'kibana/config',
+  'kibana/notify',
+  'ngRoute',
+]);
+
+module.directive('contextApp', function ContextApp() {
+  return {
+    bindToController: true,
+    controller: ContextAppController,
+    controllerAs: 'contextApp',
+    restrict: 'E',
+    scope: {
+      anchorUid: '=',
+      columns: '=',
+      indexPattern: '=',
+      predecessorCount: '=',
+      successorCount: '=',
+      sort: '=',
+      discoverUrl: '=',
+    },
+    template: contextAppTemplate,
+  };
+});
+
+function ContextAppController($scope, Private) {
+  const queryParameterActions = Private(QueryParameterActionsProvider);
+  const queryActions = Private(QueryActionsProvider);
+
+  this.state = createInitialState(this.discoverUrl);
+
+  this.actions = _.mapValues(Object.assign(
+    {},
+    queryParameterActions,
+    queryActions,
+  ), (action) => (...args) => action(this.state)(...args));
```

Comment:
I can see the argument for explicitly mentioning these few actions, but I'm not sure it would scale well for a larger number of actions. Keeping the explicit list of action DRY by mapping over them looks like the obvious thing to do here. How else would you avoid the repitition?